### ISSUE TITLE
Checking for alias variables and variability

### ISSUE THREAD

**Issue by nickbattle** [modelica/fmi-cross-check#76]
FMI Standard issue 2.0.1 included a clarification paragraph to explain the rules regarding the variability setting of collections of aliased variables. VDMCheck was enhanced to make this new check, and the following list of FMUs in the cross-check repo were found to violate the rules:

[alias_errors.txt](https://github.com/modelica/fmi-cross-check/files/3733849/alias_errors.txt)

See https://github.com/modelica/fmi-standard/issues/436

**Comment by chrbertsch** [comment]
@nickbattle : Thanks!

IMHO this is a case where it is great to have a checker that is very strict to provide a warning, but we should not remove these FMUs from the FMI 2.0 XC repository, as the clarification came only in FMI 2.0.1

It is most important that the relevant tool vendor change this for the future.
If see it right: only Dymola/CATIA and OpenModelica are involved, right? 
So I will forward this issue to Dassault (@KarlWernersson , Henrik) and OpenModelica (@sjoelund , @lochel)

**Comment by nickbattle** [comment]
@chrbertsch OK, I'll downgrade the VDMCheck for this to be a warning (ie. if this is the only problem, it will report "No errors found").  Yes, I can confirm that it is only the three tools that you mentioned.

**Comment by nickbattle** [comment]
VDMCheck 0.0.2 build 191016 now treats the new 2.0.1 errors as warnings:
```
------------------------------- ./me/win64/CATIA/R2016x/ControlledTemperature/ControlledTemperature.fmu
2.2.7 Warning: aliases of reference 16777218/Real must all be <tunable>, because of "constantVoltage.V"
2.2.7 Warning: aliases of reference 100663309/Real must all be <continuous>
2.2.7 Aliases of reference 335544321/Real must all have same unit/baseUnits
2.2.7 Invalid ScalarVariable aliasing
2.2.1 Effective ScalarVariables invalid
Errors found.
```
https://github.com/INTO-CPS-Association/FMI2-VDM-Model/releases/download/0.0.2-1/vdmcheck-0.0.2-191016-distribution.zip

**Comment by t-sommer** [comment]
If we can agree on a hard rule, I will add it to CI.

@nickbattle, can you post the code you're using to detect the issue?

**Comment by nickbattle** [comment]
@t-sommer PS. I hate to quibble, but these things come up when you express rules formally...

The rule 1, regarding settable aliases, excludes the case of having both "input" and "parameter" types in the alias set. But it says hte exclusion are variables that can be set with fmi2SetXXX. In the case of "parameter" types, this can mean that they are both "parameter" and "tunable" (rule 4 on p88 or rule 6 on p88, depending on the state).

So perhaps you could have a parameter that was not tunable and an input variable in the same alias group? In which case, what should be the variability of the group be: that of the "input" or that of the "parameter"? Or should the alias rule be for "tunable parameters" and inputs?

**Comment by HansOlsson** [comment]
I commented on the specification itself in https://github.com/modelica/fmi-standard/issues/436

However, for the cross-checker the most important issue is that the variability-text (as far as I can tell) is non-normative, and thus not something we should reject models based on - even for 2.0.1.

**Comment by nickbattle** [comment]
@HansOlsson These checks have been downgraded to warnings only now, so they will not fail an FMU by themselves (in VDMCheck).

**Comment by chrbertsch** [comment]
> @chrbertsch I cannot find the ticket you mentioned above. Can you please send me the ticket number?

Thanks, I seem not to have sent the ticket mentioned above. I have done it now: https://trac.openmodelica.org/OpenModelica/ticket/5673
@lochel : Thanks in advance for looking into this; looking forward to see OpenModelica  join the FMI XC effort ...
